File: NapakalakiJava/src/NapakalakiGame/Treasure.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author danibolanos & jomabose
 */
public class Treasure {
    private String name;
    private int bonus;
    private TreasureKind type;
    
    public Treasure(String n, int bonus, TreasureKind t){
        name = n;
        this.bonus = bonus;
        type = t;
    }
    
    public String getName(){
        return name;
    }
    
    public int getBonus(){
        return bonus;
    }
    
    public TreasureKind getType(){
        return type;
    }
    public String toString(){
        String cadena = name;
        cadena += "\nBonus: " + bonus;
        cadena += "\nTipo: " + type;
        
        return cadena;
    }
    public String getInfo(){
        String cadena=name + ", +" + bonus + ", " + type;
        return cadena;
    }
}

File: NapakalakiRuby/lib/treasure.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
  
  require_relative 'treasure_kind'

class Treasure
  
  public
  
  def initialize(n, bonus, t)
    @name = n
    @bonus = bonus
    @type = t
  end
  
  def getName
    @name
  end
  
  def getBonus
    @bonus
  end
  
  def getType
    @type
  end
  
  def to_s
    cadena = "#{@name}"
    cadena += "\nBonus: #{@bonus}"
    cadena += "\nTipo: #{@type}"
    cadena
  end
end

end
